tests/python_client/testcases/test_search.py

```diff
@@ -2667,6 +2667,41 @@ def test_search_expression_all_data_type(self, nb, nq, dim, auto_id, _async):
         assert (default_int64_field_name and default_float_field_name and default_bool_field_name) \
                in res[0][0].entity._row_data
 
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.skip(reason="issue #23646")
+    @pytest.mark.parametrize("field", ct.all_scalar_data_types[:7])
+    def test_search_expression_different_data_type(self, field):
+        """
+        target: test search expression using different supported data types
+        method: search using different supported data types
+        expected: search success
+        """
+        # 1. initialize with data
+        default_schema = cf.gen_collection_schema_all_datatype()
+        collection_w = self.init_collection_wrap(schema=default_schema)
+        collection_w = cf.insert_data(collection_w, is_all_data_type=True, insert_offset=-1500)[0]
+
+        # 2. create index and load
+        collection_w.create_index(field_name, default_index_params)
+        collection_w.load()
+
+        # 3. search
+        log.info("test_search_expression_different_data_type: Searching collection %s" % collection_w.name)
+        expression = "bool == True" if field == "bool" else f"{field} >= 1"
```

Comment:
`f"{field} >= 1"` may not trigger the issue.
It was triggered due to being out of bounds. 
for example, the range of the int8 datatype is [-128, 127], so you need to set the expression as int8 > 128 or int8 < -129.

Reply:
Great idea. I'll modify the cases.